### PR TITLE
Groom GitHub CI and move some checks from TravisCI to GitHub CI

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -54,6 +57,7 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php-version }}
+          coverage: none
```
Comment (keradus) [inline .github/workflows/ci.yaml:60]: Before:

Comment (keradus) [inline .github/workflows/ci.yaml:60]: Before:
Screenshot from 2020-11-29 02-36-36

After
Screenshot from 2020-11-29 02-52-08

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -70,6 +87,11 @@ jobs:
             composer-${{ runner.os }}-
             composer-
 
+      - name: Configure Symfony Flex
+        run: |
+            composer config extra.symfony.require ${{ matrix.execute-flex-with-symfony-version }}
+        if: "matrix.execute-flex-with-symfony-version"
```
Comment (sanmai) [inline .github/workflows/ci.yaml:93]: What do you think if `if:` would go before `run:`?

**Review by sanmai — APPROVED** [review]
Looks pretty solid to me.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (keradus) [reply .github/workflows/ci.yaml:93]: No preference

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -78,13 +100,18 @@ jobs:
           retry_wait_seconds: 30
           command: |
             composer update --optimize-autoloader --no-interaction --no-progress ${{ matrix.composer-flags }}
+            composer info -D | sort
```
Comment (julienfalque) [inline .github/workflows/ci.yaml:103]: This is redundant with output of `composer update`.

**Review by julienfalque — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (keradus) [reply .github/workflows/ci.yaml:103]: I find `-D` and `sort` useful
actually even more useful than output of `update`. maybe let's use `update -q` ?

Comment (julienfalque) [reply .github/workflows/ci.yaml:103]: That would leave us with less information than actually provided.

Comment (keradus) [reply .github/workflows/ci.yaml:103]: so... maybe let's keep both?

Comment (keradus) [reply .github/workflows/ci.yaml:101]: let's merge as-is and we can always adjust later ;)

**Pull request merged**